### PR TITLE
Bug 1882203: separate playbook to redeploy named certs only

### PR DESCRIPTION
**Pull request by vwalek**
playbook to redeploy on the named certificates with option to add the 'cafile' from the named certificates to the ca-bundle.crt

### CODE REVIEW

File: roles/openshift_named_certificates/tasks/main.yml [hunk 1]
```diff
@@ -13,6 +13,19 @@
...
+    get_attributes: false
+    get_mime: false
+  register: named_certs_dir_stat
+
+- name: Backup Named Certificates
+  command: >
+    tar -czvf /etc/origin/master-named-certificates-backup-{{ ansible_date_time.epoch }}.tgz
+    {{ named_certs_dir }}
+  when: named_certs_dir_stat.stat.exists
+
```
Comment (vwalek) [inline roles/openshift_named_certificates/tasks/main.yml:28]: Added backup option for named certificates.

The named certs are in fact backup by the certificate-backup.yml playbook as part of the /etc/origin/master directory
https://github.com/openshift/openshift-ansible/blob/release-3.11/playbooks/openshift-master/private/certificates-backup.yml#L37-L40

File: roles/openshift_named_certificates/tasks/main.yml [hunk 2]
```diff
@@ -40,10 +53,42 @@
...
     mode: 0600
   with_items: "{{ named_certificates | lib_utils_oo_collect('cafile') }}"
+
+- name: Verify if the ca-bundle contains the CA to trust the named certificate
+  command: >
+    openssl verify -CAfile /etc/origin/master/ca-bundle.crt "{{ named_certs_dir }}{{ item.certfile | basename }}"
+  with_items: "{{ named_certificates }}"
+  ignore_errors: True
+  register: ca_bundle_result
+  when: named_certificates_ca_file_defined | length > 0
```
Comment (vwalek) [inline roles/openshift_named_certificates/tasks/main.yml:73]: The task just verifies trust the of the named certificates and registers the output. If the ca-bundle.crt doesn't trust the named certs, the following task appends the cafile to the ca-bundle.

I understand that the openshift-master/redeploy-certificates will overwrite that, however, the result at the end is the same.

File: playbooks/openshift-master/private/redeploy-named-certificates.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+- import_playbook: named-certificates.yml
```
Comment (mtnbikenc) [inline playbooks/openshift-master/private/redeploy-named-certificates.yml:2]: I'm not sure I understand the usage of both private/named-certificates.yml and private/redeploy-named-certificates.yml playbooks.  Couldn't the `vars` just be added to the one play from named-certificates.yml and just use private/redeploy-named-certificates.yml?

Also, yamllint issues.

File: roles/openshift_named_certificates/tasks/main.yml [hunk 4]
```diff
@@ -13,6 +13,19 @@
     additive_facts_to_overwrite:
     - "{{ 'master.named_certificates' if overwrite_named_certs | bool else omit }}"
 
+- stat:
```
Comment (mtnbikenc) [inline roles/openshift_named_certificates/tasks/main.yml:16]: Please add `- name:` to all unnamed tasks.

File: roles/openshift_named_certificates/tasks/main.yml [hunk 5]
```diff
@@ -40,10 +53,43 @@
...
+  with_items: "{{ named_certificates }}"
+  ignore_errors: True
+  register: ca_bundle_result
+  when: named_certificates_ca_file_defined | length > 0
+
+- name: Append the custom ca to the bundle
+  block:
+  - command: mktemp -d /tmp/openshift-ansible-XXXXXX
+    register: openshift_ca_named_certificates_tmpdir
+  - copy:
```
Comment (mtnbikenc) [inline roles/openshift_named_certificates/tasks/main.yml:79]: Please add `- name:` to all unnamed tasks.

File: roles/openshift_named_certificates/tasks/main.yml [hunk 6]
```diff
@@ -40,10 +53,43 @@
...
+      dest: "{{ openshift_ca_named_certificates_tmpdir.stdout }}"
+      remote_src: true
+    with_items: "{{ named_certificates | lib_utils_oo_collect('cafile') }}"
+  - assemble:
+      src: "{{ openshift_ca_named_certificates_tmpdir.stdout }}"
+      dest: "/etc/origin/master/ca-bundle.crt"
+      mode: 0644
+      owner: root
+      group: root
+  when: not ca_bundle_result and (named_certificates_ca_file_defined | length > 0)
```
Comment (mtnbikenc) [inline roles/openshift_named_certificates/tasks/main.yml:94]: Please use list-style conditionals for `and`.
```suggestion
  when:
  - not ca_bundle_result
  - named_certificates_ca_file_defined | length > 0
```

**Review by mtnbikenc — CHANGES_REQUESTED** [review]
Please fix yamllint issues reported in unit test.

File: playbooks/openshift-master/private/redeploy-named-certificates.yml (hunk 3 above)
Comment (vwalek) [reply playbooks/openshift-master/private/redeploy-named-certificates.yml:2]: Hey,
I was going with the same practice as redeploying master certs playbook. Ok, I understand the point. So you wan't to include directly the tasks in `private/redeploy-named-certificates.yml` - and totally skip the import of `named-certificates.yml`.

I will work on that. Also I will fix the yamllint issues.

**Comment by vwalek** [comment]
sorry for so many changes - I didn't have VSCode setup correctly. Now it should be better. Thx

**Review by mtnbikenc — APPROVED** [review]
/lgtm

**Pull request merged**